create-publication-from-doi/src/test/java/no/unit/nva/doi/fetch/service/IdentityUpdaterTest.java

```diff
@@ -11,110 +12,118 @@
-        Publication updatedPublication = IdentityUpdater.enrichPublicationCreators(bareProxyClient, publication);
+        var cristinProxyClient = mock(CristinProxyClient.class);
+        var publication = createPublicationWithIdentity(new Identity());
+        var updatedPublication = IdentityUpdater.enrichPublicationCreators(cristinProxyClient, publication);
+
         assertNotNull(publication);
         assertNotNull(updatedPublication);
         assertEquals(publication, updatedPublication);
     }
 
     @Test
     public void enrichPublicationCreatorsDoesNotAlterPublicationWithoutEntityDescription() {
-        Identity identity = new Identity();
-        BareProxyClient bareProxyClient = mock(BareProxyClient.class);
-        Publication publication = createPublicationWithIdentity(identity);
+        var cristinProxyClient = mock(CristinProxyClient.class);
+        var publication = createPublicationWithIdentity(new Identity());
         publication.setEntityDescription(null);
-        Publication updatedPublication = IdentityUpdater.enrichPublicationCreators(bareProxyClient, publication);
+        var updatedPublication = IdentityUpdater.enrichPublicationCreators(cristinProxyClient, publication);
+
         assertNotNull(publication);
         assertNotNull(updatedPublication);
     }
 
+    @Test
+    public void enrichPublicationCreatorsDoesNotCrashOnPublicationWithoutContributors() {
+        var cristinProxyClient = mock(CristinProxyClient.class);
+        var publication = createPublicationWithIdentity(new Identity());
+        var entityDescriptionWithoutContributors =
+            new EntityDescription.Builder().withContributors(null).build();
+        publication.setEntityDescription(entityDescriptionWithoutContributors);
+
+        IdentityUpdater.enrichPublicationCreators(cristinProxyClient, publication);
+    }
 
     @Test
-    public void enrichPublicationCreatorsDoesUpdatePublicationCreatorWithOrcid() {
-        Identity identity = createIdentityWithOrcid();
-        BareProxyClient bareProxyClient = mock(BareProxyClient.class);
-        Publication publication = createPublicationWithIdentity(identity);
-        Publication updatedPublication = IdentityUpdater.enrichPublicationCreators(bareProxyClient, publication);
-        assertNotNull(publication);
-        assertNotNull(updatedPublication);
+    public void enrichPublicationCreatorsIgnoresExceptionsAndReturnsUnmodifiedPublication() {
+        var identity = new Identity.Builder().withOrcId(ILLEGAL_ORCID).build();
+        var publication = createPublicationWithIdentity(identity);
+        var cristinProxyClient = mock(CristinProxyClient.class);
+        doThrow(new RuntimeException()).when(cristinProxyClient).lookupIdentifierFromOrcid(any());
+        var updatedPublication = IdentityUpdater.enrichPublicationCreators(cristinProxyClient, publication);
+
+        assertEquals(publication, updatedPublication);
```

Comment:
We're not using hamcrest, either :( I guess that's a task for a new PR too

Reply:
Thought about hamcrest and shouldXXX but tried to follow the original code pattern in that class. Will update in new PR. What do you think for test names? `public void` or just `void`?